### PR TITLE
feat(app subscription): ui enhancements and functionality changes

### PR DESCRIPTION
**Pull request by nidhigarg-bmw**
## Description

- View Detail button Overlay functionality
- App filter Functionality
- Activation Overlay Refracting

## Issue

CPLP-2510
CPLP-2586
CPLP-2587
CPLP-2664

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

### CODE REVIEW

File: cx-portal/src/components/pages/AppSubscription/AppSubscriptionDetailOverlay/index.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      headline: t('content.appSubscription.detailOverlay.stepLists.thirdStep'),
+      step: 3,
+    },
+  ]
+
+  const subscriptionDetails: TableType = {
+    head: [t('content.appSubscription.detailOverlay.subscriptionDetails'), ''],
+    body: [
+      [
+        `${t('content.appSubscription.detailOverlay.appTitle')}`,
```
Comment (oyo) [inline cx-portal/src/components/pages/AppSubscription/AppSubscriptionDetailOverlay/index.tsx:69]: no need to wrap in a template string:
`t('...'),` 
should be okay. Also for following lines.

File: cx-portal/src/components/shared/templates/Subscription/index.tsx [hunk 2]
```diff
@@ -371,6 +413,26 @@ export default function Subscription({
...
+  useEffect(() => {
+    if (tabLabels.showAll) {
+      setState({ type: ActionKind.SET_SELECTED, payload: FilterType.SHOWALL })
+      setState({ type: ActionKind.SET_STATUS_ID, payload: '' })
+      setState({
+        type: ActionKind.SET_FETCH_ARGS,
+        payload: { ...fetchArgs, statusId: '' },
+      })
+    }
+    // eslint-disable-next-line
```
Comment (oyo) [inline cx-portal/src/components/shared/templates/Subscription/index.tsx:433]: no linter disable

File: cx-portal/src/features/appSubscription/appSubscriptionApiSlice.ts [hunk 3]
```diff
@@ -80,11 +121,34 @@ export const apiSlice = createApi({
...
+    }),
+    fetchSubscriptionDetail: builder.query<
+      SubscriptionDetailResponse,
+      SubscriptionDetailRequestBody
+    >({
+      query: (body) => {
+        return {
+          url: `/api/apps/${body.appId}/subscription/${body.subscriptionId}/provider`,
+        }
+      },
```
Comment (oyo) [inline cx-portal/src/features/appSubscription/appSubscriptionApiSlice.ts:146]: no need for `return` statement when arrow function has no other logic than simply return one JSON object, wrap it in round brackets. Here:
`(body) => ({ url: ... })`

**Review by oyo — CHANGES_REQUESTED** [review]
some remarks plus code smells and merge conflict

File: cx-portal/src/components/pages/AppSubscription/AppSubscriptionDetailOverlay/index.tsx (hunk 1 above)
Comment (nidhigarg-bmw) [reply cx-portal/src/components/pages/AppSubscription/AppSubscriptionDetailOverlay/index.tsx:69]: @oyo It's accepting string only but not accepting translation string. So I did in this way.

File: cx-portal/src/components/pages/AppSubscription/ActivateSubscriptionOverlay/index.tsx [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+  const [activationResponse, setActivationResponse] =
+    useState<SubscriptionActivationResponse>()
+
+  const [addUserSubscribtion] = useAddUserSubscribtionMutation()
+  const { data } = useFetchTechnicalProfilesQuery(appId)
+
+  const addInputURL = (value: string) => {
+    setInputURL(value)
+    setURLErrorMessage(
+      !Patterns.URL.test(value.trim())
```
Comment (oyo) [inline cx-portal/src/components/pages/AppSubscription/ActivateSubscriptionOverlay/index.tsx:81]: we have a more expressive function for this:
`isURL = (expr: string) => Patterns.URL.test(expr)`
https://github.com/eclipse-tractusx/portal-frontend/blob/dev/cx-portal/src/types/Patterns.ts#L93

File: cx-portal/src/components/pages/AppSubscription/ActivateSubscriptionOverlay/index.tsx [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+  const addTentantURL = async () => {
+    setLoading(true)
+    try {
+      const subscriptionData = await addUserSubscribtion({
+        requestId: subscriptionId,
+        offerUrl: inputURL,
+      }).unwrap()
+      setActivationResponse(subscriptionData)
+    } catch (error) {
+      console.log(error)
```
Comment (oyo) [inline cx-portal/src/components/pages/AppSubscription/ActivateSubscriptionOverlay/index.tsx:96]: just a question - is it safe to ignore the error and log it to console or should the user know about it as a Snackbar notification? in that case better use NotifiyService error function

**Review by oyo — CHANGES_REQUESTED** [review]
one tiny change and a question

**Review by oyo — APPROVED** [review]

**Pull request merged**